File: lesson_8/app/src/main/java/com/kalashnyk/denys/gmapsapp/di/component/ViewModelComponent.java
package com.kalashnyk.denys.gmapsapp.di.component;

import com.kalashnyk.denys.gmapsapp.di.module.ViewModelModule;
import com.kalashnyk.denys.gmapsapp.di.scope.ViewModelScope;
import com.kalashnyk.denys.gmapsapp.presentation.activities.main.MainActivity;

import dagger.Component;

@ViewModelScope
@Component(modules = {ViewModelModule.class}, dependencies = {RepositoryComponent.class})
public interface ViewModelComponent {
    void inject(MainActivity activity);
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/di/component/ViewModelComponent.kt
package com.kalashnyk.denys.kotlinsample.di.component

import com.kalashnyk.denys.kotlinsample.di.module.ViewModelModule
import com.kalashnyk.denys.kotlinsample.di.scope.ViewModelScope
import com.kalashnyk.denys.kotlinsample.presentation.activities.detail.DetailActivity
import com.kalashnyk.denys.kotlinsample.presentation.activities.main.MainActivity
import dagger.Component

@ViewModelScope
@Component(modules = [ViewModelModule::class], dependencies = [RepositoryComponent::class])
interface ViewModelComponent {
    fun inject(activity: MainActivity)
    fun inject(activity: DetailActivity)
}